src/Compiler/Options.hs

```diff
@@ -632,6 +644,18 @@ getOptions extra
        args <- getArgs
        processOptions flagsNull (env ++ words extra ++ args)
 
+updateFlagsFromArgs :: Flags -> String -> Maybe Flags
+updateFlagsFromArgs flags0 args =
```

Comment:
This is used for parsing additional arguments from the language server's run command.